src/Voron/Impl/LowLevelTransaction.cs

```diff
@@ -1046,13 +1047,22 @@ public void FreePage(long pageNumber)
             }
         }
 
+        private static readonly ObjectPool<CompactKey> _sharedCompactKeyPool = new ( () => new CompactKey() );
+
         public CompactKey AcquireCompactKey()
         {
-            // Originally the low level transaction would allow to handle the reuse of compact keys.
-            // However, the implementation of reuse has been creating issues when indexing. The ability
-            // to reuse keys will be disable until we are able to resolve the underlying cause for it.
-            // https://issues.hibernatingrhinos.com/issue/RavenDB-20143
-            return new CompactKey(this);
+            var key = _sharedCompactKeyPool.Allocate();
+            key.Initialize(this);
+            return key;
+        }
+
+        public void ReleaseCompactKey(ref CompactKey key)
```

Comment:
shouldn't you handle this already being null?

Reply:
If that happens, it should fail fast. It's a big bug.

Reply:
In general, we aren't going to consider disposing twice a sign of a problem.
That can happen for a variety of reasons
For example:

![image](https://github.com/ravendb/ravendb/assets/116915/a19b86f8-5d4e-46bc-9684-212e77523376)

Reply:
In this case, we have a batch, and we hit a scenario where we have to flush it, then dispose.
The flush already dispose the key, but the dispose does that as well

One can also thing of other scenarios, where we have a `CompactKeyCacheScope` that wasn't init (maybe an error) but is called from the dispose.